Python/_warnings.c

```diff
@@ -520,14 +525,20 @@ warn_explicit(PyObject *category, PyObject *message,
         goto cleanup;
     }
 
+    if (_PyUnicode_EqualToASCIIString(action, "ignore")) {
+        goto return_none;
+    }
+
     /* Store in the registry that we've been here, *except* when the action
        is "always". */
     rc = 0;
     if (!_PyUnicode_EqualToASCIIString(action, "always")) {
         if (registry != NULL && registry != Py_None &&
-                PyDict_SetItem(registry, key, Py_True) < 0)
+                PyDict_SetItem(registry, key, Py_True) < 0) {
```

Comment:
If you want to add braces, place the opening brace after a multiline condition on a separate line.

Reply:
The PEP 7 only shows "if (test) {", it says nothing about multiline condition. Until a newline is required by the PEP 7, I prefer to put "{" on the same line.

Reply:
```
* When you break a long expression at a binary operator, the
  operator goes at the end of the previous line, and braces should be
  formatted as shown.  E.g.::

      if (type->tp_dictoffset != 0 && base->tp_dictoffset == 0 &&
          type->tp_dictoffset == b_size &&
          (size_t)t_size == b_size + sizeof(PyObject *))
      {
          return 0; /* "Forgive" adding a __dict__ only */
      }
```

Reply:
Oh nice, it's specified in the PEP 7. In that case, ok to reformat the code as the PEP 7 requires ;-)